vendor/bat-native-ledger/src/bat/ledger/internal/core/bat_ledger_context.cc

```diff
@@ -7,44 +7,60 @@
 
 #include <utility>
 
+#include "bat/ledger/internal/constants.h"
 #include "bat/ledger/internal/ledger_impl.h"
+#include "bat/ledger/ledger.h"
 #include "bat/ledger/ledger_client.h"
+#include "bat/ledger/option_keys.h"
+#include "bat/ledger/public/interfaces/ledger.mojom.h"
 
 namespace ledger {
 
 namespace {
 
-size_t g_next_component_key = 0;
+BATLedgerContext::Environment GetEnvironment() {
+  switch (ledger::_environment) {
+    case mojom::Environment::DEVELOPMENT:
+      return BATLedgerContext::Environment::kDevelopment;
+    case mojom::Environment::STAGING:
+      return BATLedgerContext::Environment::kStaging;
+    case mojom::Environment::PRODUCTION:
+      return BATLedgerContext::Environment::kProduction;
```

Comment:
```cpp
default:
  NOTREACHED();
```

Reply:
Would it be better to leave the `default` case out so that we get a compile time error if an env was added instead of a runtime (debug) exception?

Reply:
Good point, agreed!